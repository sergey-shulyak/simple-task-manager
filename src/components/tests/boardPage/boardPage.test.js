import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { columns, tickets } from '../testData';

import BoardPage from '../../boardPage';

const getProps = overrides => ({
  board: { id: '42', title: 'Test board', columns },
  tickets,
  fetchBoard: jest.fn(),
  fetchTickets: jest.fn(),
  match: { params: { id: '42' } },
  ...overrides
});

describe('BoardListPage', () => {
  it('Should render correctly', () => {
    const props = getProps();
    const wrapper = shallow(<BoardPage {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('Should call fetchBoard on componentWillMount with empty Board', () => {
    const props = getProps({ board: {} });
    const wrapper = shallow(<BoardPage {...props} />);

    wrapper.simulate('componentWillMount');

    expect(props.fetchBoard.mock.calls.length).toBe(1);
  });

  it('Should call fetchTickets on componentWillMount', () => {
    const props = getProps();
    const wrapper = shallow(<BoardPage {...props} />);

    wrapper.simulate('componentWillMount');

    expect(props.fetchTickets.mock.calls.length).toBe(1);
  });
});
