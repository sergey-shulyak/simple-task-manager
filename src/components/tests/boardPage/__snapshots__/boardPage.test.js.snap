// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoardListPage Should render correctly 1`] = `
<div
  className="board-page"
>
  <Link
    className="board-page__back-link"
    replace={false}
    to="/boards"
  >
    Go back
  </Link>
  <Board
    className="board-page__board"
    columns={
      Array [
        Object {
          "id": "1",
          "title": "To do",
        },
        Object {
          "id": "2",
          "title": "In progress",
        },
        Object {
          "id": "3",
          "title": "Done",
        },
      ]
    }
    id="42"
    tickets={
      Array [
        Object {
          "boardId": "42",
          "columnId": "1",
          "id": "1",
          "title": "Test ticket 1",
          "type": "task",
        },
        Object {
          "boardId": "42",
          "columnId": "2",
          "id": "2",
          "title": "Test ticket 1",
          "type": "task",
        },
        Object {
          "boardId": "42",
          "columnId": "3",
          "id": "3",
          "title": "Test ticket 1",
          "type": "task",
        },
      ]
    }
    title="Test board"
  />
</div>
`;
